### PR TITLE
Separate out feature tracking module

### PR DESCRIPTION
**Pull request by zephyrdaddy**
@johannes-graeter 
I separated the tracking processor into a separate package so that it can be used in the main program. 
I need to check if it runs without any problem before merging it, but would you please take a look at it if it looks reasonable?

### CODE REVIEW

File: viso_feature_tracking_ros_tool/src/viso_feature_tracking_node/viso_feature_tracking_interface.cpp [hunk 1]
```diff
@@ -62,106 +57,20 @@ void VisoFeatureTrackingInterface::process(const sensor_msgs::ImageConstPtr& inp
...
-    cv::GaussianBlur(cv_bridge_ptr->image,
-                     cv_bridge_ptr->image,
-                     cv::Size(viso_params.blur_size, viso_params.blur_size),
-                     viso_params.blur_sigma);
-
-    tracker->pushBack(cv_bridge_ptr->image);
-    tracker->getTracklets(this->tracklets_, 0);
-
-    this->timestamps_.push_front(input->header.stamp);
+    tracker_module_->process(cv_bridge_ptr->image, input->header.stamp);
```
Comment (zephyrdaddy) [inline viso_feature_tracking_ros_tool/src/viso_feature_tracking_node/viso_feature_tracking_interface.cpp:63]: This is the main part I took out from the original code. All the actual processing happens in the tracker_module_.

Comment (johannes-graeter) [reply viso_feature_tracking_ros_tool/src/viso_feature_tracking_node/viso_feature_tracking_interface.cpp:63]: Great!

File: viso_feature_tracking_ros_tool/src/viso_feature_tracking_node/viso_feature_tracking_interface.cpp [hunk 2]
```diff
@@ -62,106 +57,20 @@ void VisoFeatureTrackingInterface::process(const sensor_msgs::ImageConstPtr& inp
...
-    publish(this->timestamps_);
+    publish();
 
-    // cut timestamps
-    if (static_cast<int>(timestamps_.size()) > max_size_timestamps_) {
-        timestamps_.pop_back();
-    }
-}
-
-MatchesMsg convert_tracklets_to_matches_msg(feature_tracking::TrackletVector m,
```
Comment (johannes-graeter) [inline viso_feature_tracking_ros_tool/src/viso_feature_tracking_node/viso_feature_tracking_interface.cpp:95]: its nice, that the timestamps are now encalsuplated by the tracker module!

**Review by johannes-graeter — APPROVED** [review]
Changes are great! Thanks

**Pull request merged**